static/js/apps/eval_retrieval_generation/sxs/eval_list.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Component to show the full list of SxS evaluation queries */
+
+import React, { useContext, useState } from "react";
+import { Button, Modal } from "reactstrap";
+
+import { AllQuery } from "../types";
+import { SessionContext } from "./context";
+import { getRatedQueryIds } from "./data_store";
+
+interface EvalListPropType {
+  leftSheetId: string;
+  rightSheetId: string;
+  sessionId: string;
+  sortedQueryIds: number[];
+  allQuery: AllQuery;
+}
+
+export function EvalList(props: EvalListPropType): JSX.Element {
+  const { sessionQueryId, setSessionQueryId } = useContext(SessionContext);
+
+  const [modalOpen, setModalOpen] = useState(false);
+
+  // Keys are query IDs and values are true if query eval is complete.
+  const [queryCompletionStatus, setQueryCompletionStatus] = useState<
+    Record<number, boolean>
+  >({});
+
+  const [completedCount, setCompletedCount] = useState<number>(0);
+
+  const openModal = () => {
+    setModalOpen(true);
+    getRatedQueryIds(
+      props.leftSheetId,
+      props.rightSheetId,
+      props.sessionId
+    ).then((completedIds) => {
+      let updatedCompletedCount = 0;
+      const updatedCompletionStatus: Record<number, boolean> = {};
+      props.sortedQueryIds.forEach((queryId) => {
+        const completed = completedIds.includes(queryId);
+        updatedCompletionStatus[queryId] = completed;
+        if (completed) updatedCompletedCount++;
+      });
+      setQueryCompletionStatus(updatedCompletionStatus);
+      setCompletedCount(updatedCompletedCount);
+    });
+  };
+
+  return (
```

Comment:
IMO there are enough substantive differences that it would add significant complexity to use the same component. See [diff](https://editor.mergely.com/7kGy7n0g) (also to summarize some other differences: no user evals only checkbox, no user email column, no need to set feedback stage or call ID, query IDs are a separate prop from queries).

Reply:
whoa that's a cool tool! ok that's fair